build.proj

```diff
@@ -34,6 +35,26 @@
   <Target Name="BuildTraversalBuildDependencies"
           DependsOnTargets="$(TraversalBuildDependencies)" />
 
+  <Target Name="CreateHostMachineInfoFile">
+    <GetTargetMachineInfo>
+      <Output PropertyName="HostMachineRid" TaskParameter="RuntimeIdentifier" />
```

Comment:
Why not capture TargetOS as OsGroup here instead of parsing the HostMachineRid in dir.props?

Reply:
Because none of those properties are "good enough".

https://github.com/dotnet/buildtools/blob/9714ddaaef88b530a9054ca6f99687fc090fd057/src/Microsoft.DotNet.Build.Tasks/GetTargetMachineInfo.cs#L41-L50

TargetOS only gets set to `Windows_NT`, not `win10` or `win7`.  It also just says `Linux`, not `ubuntu.14.04` or `centos.7.1`.

Reply:
Note: corefx uses a completely different BuildTools Task for this:

https://github.com/dotnet/buildtools/blob/03988127f489bcd1cf58d71131ac628620d53de9/src/Microsoft.DotNet.Build.Tasks/GenerateConfigurationProps.cs#L135-L155

Which does the same parsing and sets the `RuntimeOS` property.